### ISSUE TITLE
Alternate style tag: usage of microformat

### ISSUE THREAD

**Issue by iherman** [w3c/epub-specs#809]
(I guess this may just rumbling at this point, because it is too late to change that...)

TL;DR: I am surprised that the microformat approach is used to define the alternate styles, i.e., predefined values for the `@class` attribute. I would have preferred to define, for example, a set of additional `@rel` values instead that could be used the same way and with the same semantics as the current microformat terms.

To avoid misunderstandings, this is not some sort of an opposition to microformats as a matter of principle. The problem I see is a discrepancy with the rest of the EPUB 3.1 specification. EPUB 3.1 has gone out of its way to use distinct attributes for such semantic information. The usage of `@epub:type` in the current EPUB document for semantic inflection (or the possible replacement thereof by the ARIA `@role` attribute in the future) is a typical example. The (re-)usage of the`@rel` attribute for the `link` element of the package is another one. Using a different mechanism for additional semantics of a specific feature looks highly inconsistent. Both of the aforementioned mechanism could have been used to tag alternate style sheets.

There may be backward compatibility issues, but it could still be possible to keep the usage of `@class` as obsolete, and introduce the same values using, for example, the `@rel` attribute...

**Comment by mattgarrish** [comment]
I'd suggest this belongs in W3C. IDPF published it, but it doesn't actually have anything specifically to do with EPUB.

**Comment by iherman** [comment]
Well... It is not a W3C document, for sure. So yes, on long term, this may be a work to be done at W3C. However, as it stands now, it is listed in the suite of EPUB3.1, it is even part of the generated EPUB version:-).

I am not sure whether this was part of EPUB3.01. If so, then there may be backward compatibility issues that dictate not to do anything with it now, and revisit the issue for later versions. But if this is a new document, then...

**Comment by mattgarrish** [comment]
I don't know the full history of how it came about, but it was written by Elika at the same time that we were doing the 3.0 revision. It wasn't done by the working group, per se, although that's somewhat immaterial since it gets WG blessing by being published. It was then published by IDPF and is referenced normatively in content documents as a way of attaching these kinds of alternate style sheets. I think it was born out of the Enhanced Global Language Support Group.

I've always been kind of uneasy with it being in the epub family, but it's published by IDPF and a normative reference, which are the two key qualifiers. It's an HTML technology at heart, though.

There are already implementations, as there was a request earlier to require the reading system to expose all available alternates. I recall Google indicating that they use it, and looking at epubtest.org there appears to be a lot of support for daytime style sheets (but the data is incomplete and I'm wondering how so many RSes can pass daytime but fail nighttime). If we change now, we are going to force changes to existing implementations.

It might be better to defer and take up again in BFF/EPUB4.

**Comment by iherman** [comment]
> I don't know the full history of how it came about, but it was written by Elika at the same time that we were doing the 3.0 revision. It wasn't done by the working group, per se, although that's somewhat immaterial since it gets WG blessing by being published. It was then published by IDPF and is referenced normatively in content documents as a way of attaching these kinds of alternate style sheets. I think it was born out of the Enhanced Global Language Support Group.
> I've always been kind of uneasy with it being in the epub family, but it's published by IDPF and a normative reference, which are the two key qualifiers. It's an HTML technology at heart, though.
> 
> There are already implementations, as there was a request earlier to require the reading system to expose all available alternates. I recall Google indicating that they use it, and looking at epubtest.org there appears to be a lot of support for daytime style sheets (but the data is incomplete and I'm wondering how so many RSes can pass daytime but fail nighttime). If we change now, we are going to force changes to existing implementations.
> 
> O.k.
> It might be better to defer and take up again in BFF/EPUB4.
> 
> That is fine with me. 

I wonder, however, whether it is possible (and worth it) to put a note somewhere, that future versions of EPUB may consider a better integration with the rest of the technologies insofar as streamlining the semantic extension formalism. Just some sort of advanced warning… Just an idea...

**Comment by laudrain** [comment]
+1 for a note for advanced warning.

**Comment by mattgarrish** [comment]
-1 to a note.

Saying what future specifications might do is awkward. It won't change anything in the here and now, and may never come to pass.

**Comment by iherman** [comment]
I will not fight for this :-) I'm ok closing the issue with no further steps.

**Comment by mattgarrish** [comment]
Let's defer. I think it's worth keeping open for future discussion.

It still feels to me like this should be in the web technology sphere, otherwise it's just a custom epub implementation that's not obviously wrapped up in epub trappings. Much like the ssml attributes.

**Comment by murata2makoto** [comment]
This was done in EGLS, although I do not remember serious discussions on the use of the class attribute.

Apparently, this mechanism is heavily used.  Let's defer.

**Comment by dauwhe** [comment]
I believe this issue is now moot, since we removed formal mention of alternate style sheets in 3.2. See #1035.
